cli/operations.py

```diff
@@ -17,62 +17,78 @@
 
 azure_storage_client = None
 
+
 class ImageLimitException(Exception):
     pass
 
 
 def supported_file_type(file_name):
+    if file_name.startswith('.'):
+        return False
+
     file_suffix = pathlib.Path(file_name).suffix.lower()
     if file_suffix in ['.png', '.jpg', '.jpeg', '.gif']:
         return True
     else:
         return False
 
 
+# Somewhat of a hack to remove the folder name passed on the
+# command line from the file path to make blob names more sane.
+def strip_path_prefix(folder_name, file_path):
+    folder_name_str = str(folder_name)
+    file_path_str = str(file_path)
+
+    stripped_path = file_path_str.replace(folder_name_str, "")
+
+    if stripped_path.startswith('/'):
+        return pathlib.Path(stripped_path[1:])
+
+    return pathlib.Path(stripped_path)
+
+
 # TODO We should create the container if it does not exist
 def onboard_folder(config, folder_name):
     blob_storage = BlobStorage.get_azure_storage_client(config)
-    uri = 'https://' + config.get("storage_account") + '.blob.core.windows.net/' + config.get("storage_container") + '/'
-    functions_url = config.get('url') + '/api/onboarding'
     user_name = config.get("tagging_user")
-    images = []
+    onboarding_files = []
+
+    print("Walking file system")
 
-    for image in os.listdir(folder_name):
-        if not supported_file_type(image):
+    for (root, dirs, files) in os.walk(folder_name):
+        # no files at this level.
+        if len(files) == 0:
             continue
 
-        local_path = os.path.join(folder_name, image)
-        print('Uploading image ' + image)
+        for file_name in files:
+            if not supported_file_type(file_name):
+                continue
+
+            relative_path = os.path.join(root, file_name)
+            onboarding_files.append(relative_path)
+
+    if len(onboarding_files) == 0:
+        print(f'Could not find any valid files to upload')
+        return
+
+    for blob_path in onboarding_files:
+        stripped_path = strip_path_prefix(folder_name, blob_path)
+        print("Uploading " + str(blob_path))
 
-        # Upload the created file, use image name for the blob name
         blob_storage.create_blob_from_path(
```

Comment:
@bnookala if we are making changes here how about adding blob metadata like [here] (https://github.com/CatalystCode/active-learning-detect/blob/a8a992231786fdaa43384119999376676b87dbcc/functions/pipeline/onboardqueueproccessor/__init__.py#L83) ?

Reply:
done